packages/core/test/features/st-custom-selector.spec.ts

```diff
@@ -59,17 +60,15 @@ describe('features/st-custom-selector', () => {
         shouldReportNoDiagnostics(meta);
     });
     it('should handle unknown custom selector', () => {
-        // ToDo: remove diagnostic once experimentalSelectorInference is the default
-        // it will fallback to pseudo-class transform and then to unknown pseudo-class
         testStylableCore(`
             /* @analyze-error(in custom) word(:--unknown) ${customSelectorDiagnostics.UNKNOWN_CUSTOM_SELECTOR(
                 ':--unknown'
             )} */
             @custom-selector :--x .before:--unknown.after;
 
             /* 
-                @transform-error(in selector) word(:--unknown)  ${customSelectorDiagnostics.UNKNOWN_CUSTOM_SELECTOR(
-                    ':--unknown'
+                @transform-error(in selector) word(--unknown)  ${pseudoClassDiagnostics.UNKNOWN_STATE_USAGE(
```

Comment:
what's behind this change?

Reply:
before the switch, inline custom selectors were transformed away before transformation process, and failed early, now they are transformed inline the transformation process, so they don't fail on unknown custom selector, but on unknown pseudo-state. 

And although you can't currently create a custom state with an intial `--`, I would like to add it in a future change.